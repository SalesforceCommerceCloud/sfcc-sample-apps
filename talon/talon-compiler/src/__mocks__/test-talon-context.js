const path = require('path');
const { startContext, getContext, endContext } = require('../context/context-service');
const crypto = require("crypto");

jest.mock('../utils/observable-folder-hash');

const DEFAULT_OPTIONS = {
    templateDir: path.resolve(__dirname, '../__tests__/fixtures'),
    basePath: '/basePath',
    outputDir: '/'
};

async function startTestContext(options = {}) {
    return startContext({
        ...DEFAULT_OPTIONS,
        ...options
    }).then(context => {
        context.versionKey = options.versionKey || crypto.randomBytes(16).toString("hex");
    });
}

function updateTestContext(options) {
    Object.assign(getContext(), options);
}

module.exports = {
    startTestContext,
    getTestContext: getContext,
    updateTestContext,
    endTestContext: endContext,
    DEFAULT_OPTIONS
};
