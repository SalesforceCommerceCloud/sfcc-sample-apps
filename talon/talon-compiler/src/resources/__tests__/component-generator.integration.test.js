import ComponentGenerator from '../component-generator';
import { startTestContext, endTestContext } from 'test-talon-context';

beforeEach(() => {
    return startTestContext();
});

afterEach(() => {
    endTestContext();
});

const generator = new ComponentGenerator();

jest.setTimeout(process.env.JEST_TIMEOUT || 30000);

describe('component-generator [integration]', () => {
    it('generates component resource', async () => {
        return generator.get('component://x/cmp1').then((staticResource) => {
            expect(staticResource).toMatchSnapshot();
        });
    });
});

