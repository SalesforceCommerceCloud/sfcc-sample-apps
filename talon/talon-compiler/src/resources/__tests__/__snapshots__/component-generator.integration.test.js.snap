// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component-generator [integration] generates component resource 1`] = `
StaticResource {
  "contents": Object {
    "compat": "Talon.moduleRegistry.define('x/cmp1', ['proxy-compat/setKey', 'proxy-compat/callKey2', 'lwc', '@babel/runtime/helpers/classCallCheck', '@babel/runtime/helpers/possibleConstructorReturn', '@babel/runtime/helpers/getPrototypeOf', '@babel/runtime/helpers/inherits', 'a'], function (__setKey, __callKey2, lwc, _classCallCheck, _possibleConstructorReturn, _getPrototypeOf, _inherits, a) { 'use strict';

    __setKey = __setKey && __setKey.hasOwnProperty('default') ? __setKey['default'] : __setKey;
    __callKey2 = __callKey2 && __callKey2.hasOwnProperty('default') ? __callKey2['default'] : __callKey2;
    _classCallCheck = _classCallCheck && _classCallCheck.hasOwnProperty('default') ? _classCallCheck['default'] : _classCallCheck;
    _possibleConstructorReturn = _possibleConstructorReturn && _possibleConstructorReturn.hasOwnProperty('default') ? _possibleConstructorReturn['default'] : _possibleConstructorReturn;
    _getPrototypeOf = _getPrototypeOf && _getPrototypeOf.hasOwnProperty('default') ? _getPrototypeOf['default'] : _getPrototypeOf;
    _inherits = _inherits && _inherits.hasOwnProperty('default') ? _inherits['default'] : _inherits;
    a = a && a.hasOwnProperty('default') ? a['default'] : a;

    function stylesheet(hostSelector, shadowSelector, nativeShadow) {
      return \\"\\\\n\\" + (nativeShadow ? \\":host {color: red;}\\" : hostSelector + \\" {color: red;}\\") + \\"\\\\n\\";
    }

    var _implicitStylesheets = [stylesheet];

    function tmpl($api, $cmp, $slotset, $ctx) {
      var api_text = $api._ES5ProxyType ? $api.get(\\"t\\") : $api.t,
          api_element = $api._ES5ProxyType ? $api.get(\\"h\\") : $api.h;
      return [api_element(\\"h2\\", {
        key: 2
      }, [api_text(\\"Cmp 1\\")])];
    }

    var _tmpl = lwc.registerTemplate(tmpl);

    __setKey(tmpl, \\"stylesheets\\", []);

    if (_implicitStylesheets) {
      __callKey2((tmpl._ES5ProxyType ? tmpl.get(\\"stylesheets\\") : tmpl.stylesheets).push, \\"apply\\", tmpl._ES5ProxyType ? tmpl.get(\\"stylesheets\\") : tmpl.stylesheets, _implicitStylesheets);
    }

    __setKey(tmpl, \\"stylesheetTokens\\", {
      hostAttribute: \\"x-cmp1_cmp1-host\\",
      shadowAttribute: \\"x-cmp1_cmp1\\"
    });

    var b = window._ES5ProxyType ? window.get(\\"B\\") : window[\\"B\\"];

    var c = window._ES5ProxyType ? window.get(\\"C\\") : window[\\"C\\"];

    a();
    b();
    c();

    var Cmp1 =
    /*#__PURE__*/
    function (_LightningElement) {
      _inherits(Cmp1, _LightningElement);

      function Cmp1() {
        _classCallCheck(this, Cmp1);

        return _possibleConstructorReturn(this, __callKey2(_getPrototypeOf(Cmp1), \\"apply\\", this, arguments));
      }

      return Cmp1;
    }(lwc.LightningElement);

    var cmp1 = lwc.registerComponent(Cmp1, {
      tmpl: _tmpl
    });

    return cmp1;

});
",
    "dev": "Talon.moduleRegistry.define('x/cmp1', ['lwc', 'a'], function (lwc, a) { 'use strict';

    a = a && a.hasOwnProperty('default') ? a['default'] : a;

    function stylesheet(hostSelector, shadowSelector, nativeShadow) {
      return \\"\\\\n\\" + (nativeShadow ? (\\":host {color: red;}\\") : (hostSelector + \\" {color: red;}\\")) + \\"\\\\n\\";
    }
    var _implicitStylesheets = [stylesheet];

    function tmpl($api, $cmp, $slotset, $ctx) {
      const {
        t: api_text,
        h: api_element
      } = $api;
      return [api_element(\\"h2\\", {
        key: 2
      }, [api_text(\\"Cmp 1\\")])];
    }

    var _tmpl = lwc.registerTemplate(tmpl);
    tmpl.stylesheets = [];

    if (_implicitStylesheets) {
      tmpl.stylesheets.push.apply(tmpl.stylesheets, _implicitStylesheets);
    }
    tmpl.stylesheetTokens = {
      hostAttribute: \\"x-cmp1_cmp1-host\\",
      shadowAttribute: \\"x-cmp1_cmp1\\"
    };

    var b = window[\\"B\\"];

    var c = window[\\"C\\"];

    a();
    b();
    c();

    class Cmp1 extends lwc.LightningElement {}

    var cmp1 = lwc.registerComponent(Cmp1, {
      tmpl: _tmpl
    });

    return cmp1;

});
",
    "prod": "Talon.moduleRegistry.define(\\"x/cmp1\\",[\\"lwc\\",\\"a\\"],function(e,t){\\"use strict\\";t=t&&t.hasOwnProperty(\\"default\\")?t.default:t;var s=[function(e,t,s){return\\"\\\\n\\"+(s?\\":host {color: red;}\\":e+\\" {color: red;}\\")+\\"\\\\n\\"}];function n(e,t,s,n){const{t:r,h:o}=e;return[o(\\"h2\\",{key:2},[r(\\"Cmp 1\\")])]}var r=e.registerTemplate(n);n.stylesheets=[],s&&n.stylesheets.push.apply(n.stylesheets,s),n.stylesheetTokens={hostAttribute:\\"x-cmp1_cmp1-host\\",shadowAttribute:\\"x-cmp1_cmp1\\"};var o=window.B,l=window.C;t(),o(),l();return e.registerComponent(class extends e.LightningElement{},{tmpl:r})});",
    "prod_compat": "Talon.moduleRegistry.define(\\"x/cmp1\\",[\\"proxy-compat/setKey\\",\\"proxy-compat/callKey2\\",\\"lwc\\",\\"@babel/runtime/helpers/classCallCheck\\",\\"@babel/runtime/helpers/possibleConstructorReturn\\",\\"@babel/runtime/helpers/getPrototypeOf\\",\\"@babel/runtime/helpers/inherits\\",\\"a\\"],function(e,t,r,s,o,n,l,a){\\"use strict\\";e=e&&e.hasOwnProperty(\\"default\\")?e.default:e,t=t&&t.hasOwnProperty(\\"default\\")?t.default:t,s=s&&s.hasOwnProperty(\\"default\\")?s.default:s,o=o&&o.hasOwnProperty(\\"default\\")?o.default:o,n=n&&n.hasOwnProperty(\\"default\\")?n.default:n,l=l&&l.hasOwnProperty(\\"default\\")?l.default:l,a=a&&a.hasOwnProperty(\\"default\\")?a.default:a;var p=[function(e,t,r){return\\"\\\\n\\"+(r?\\":host {color: red;}\\":e+\\" {color: red;}\\")+\\"\\\\n\\"}];function u(e,t,r,s){var o=e._ES5ProxyType?e.get(\\"t\\"):e.t;return[(e._ES5ProxyType?e.get(\\"h\\"):e.h)(\\"h2\\",{key:2},[o(\\"Cmp 1\\")])]}var y=r.registerTemplate(u);e(u,\\"stylesheets\\",[]),p&&t((u._ES5ProxyType?u.get(\\"stylesheets\\"):u.stylesheets).push,\\"apply\\",u._ES5ProxyType?u.get(\\"stylesheets\\"):u.stylesheets,p),e(u,\\"stylesheetTokens\\",{hostAttribute:\\"x-cmp1_cmp1-host\\",shadowAttribute:\\"x-cmp1_cmp1\\"});var h=window._ES5ProxyType?window.get(\\"B\\"):window.B,i=window._ES5ProxyType?window.get(\\"C\\"):window.C;a(),h(),i();var d=function(e){function r(){return s(this,r),o(this,t(n(r),\\"apply\\",this,arguments))}return l(r,e),r}(r.LightningElement);return r.registerComponent(d,{tmpl:y})});",
    "prod_debug": "Talon.moduleRegistry.define('x/cmp1', ['lwc', 'a'], function (lwc, a) { 'use strict';

    a = a && a.hasOwnProperty('default') ? a['default'] : a;

    function stylesheet(hostSelector, shadowSelector, nativeShadow) {
      return \\"\\\\n\\" + (nativeShadow ? (\\":host {color: red;}\\") : (hostSelector + \\" {color: red;}\\")) + \\"\\\\n\\";
    }
    var _implicitStylesheets = [stylesheet];

    function tmpl($api, $cmp, $slotset, $ctx) {
      const {
        t: api_text,
        h: api_element
      } = $api;
      return [api_element(\\"h2\\", {
        key: 2
      }, [api_text(\\"Cmp 1\\")])];
    }

    var _tmpl = lwc.registerTemplate(tmpl);
    tmpl.stylesheets = [];

    if (_implicitStylesheets) {
      tmpl.stylesheets.push.apply(tmpl.stylesheets, _implicitStylesheets);
    }
    tmpl.stylesheetTokens = {
      hostAttribute: \\"x-cmp1_cmp1-host\\",
      shadowAttribute: \\"x-cmp1_cmp1\\"
    };

    var b = window[\\"B\\"];

    var c = window[\\"C\\"];

    a();
    b();
    c();

    class Cmp1 extends lwc.LightningElement {}

    var cmp1 = lwc.registerComponent(Cmp1, {
      tmpl: _tmpl
    });

    return cmp1;

});
",
    "prod_debug_compat": "Talon.moduleRegistry.define('x/cmp1', ['proxy-compat/setKey', 'proxy-compat/callKey2', 'lwc', '@babel/runtime/helpers/classCallCheck', '@babel/runtime/helpers/possibleConstructorReturn', '@babel/runtime/helpers/getPrototypeOf', '@babel/runtime/helpers/inherits', 'a'], function (__setKey, __callKey2, lwc, _classCallCheck, _possibleConstructorReturn, _getPrototypeOf, _inherits, a) { 'use strict';

    __setKey = __setKey && __setKey.hasOwnProperty('default') ? __setKey['default'] : __setKey;
    __callKey2 = __callKey2 && __callKey2.hasOwnProperty('default') ? __callKey2['default'] : __callKey2;
    _classCallCheck = _classCallCheck && _classCallCheck.hasOwnProperty('default') ? _classCallCheck['default'] : _classCallCheck;
    _possibleConstructorReturn = _possibleConstructorReturn && _possibleConstructorReturn.hasOwnProperty('default') ? _possibleConstructorReturn['default'] : _possibleConstructorReturn;
    _getPrototypeOf = _getPrototypeOf && _getPrototypeOf.hasOwnProperty('default') ? _getPrototypeOf['default'] : _getPrototypeOf;
    _inherits = _inherits && _inherits.hasOwnProperty('default') ? _inherits['default'] : _inherits;
    a = a && a.hasOwnProperty('default') ? a['default'] : a;

    function stylesheet(hostSelector, shadowSelector, nativeShadow) {
      return \\"\\\\n\\" + (nativeShadow ? \\":host {color: red;}\\" : hostSelector + \\" {color: red;}\\") + \\"\\\\n\\";
    }

    var _implicitStylesheets = [stylesheet];

    function tmpl($api, $cmp, $slotset, $ctx) {
      var api_text = $api._ES5ProxyType ? $api.get(\\"t\\") : $api.t,
          api_element = $api._ES5ProxyType ? $api.get(\\"h\\") : $api.h;
      return [api_element(\\"h2\\", {
        key: 2
      }, [api_text(\\"Cmp 1\\")])];
    }

    var _tmpl = lwc.registerTemplate(tmpl);

    __setKey(tmpl, \\"stylesheets\\", []);

    if (_implicitStylesheets) {
      __callKey2((tmpl._ES5ProxyType ? tmpl.get(\\"stylesheets\\") : tmpl.stylesheets).push, \\"apply\\", tmpl._ES5ProxyType ? tmpl.get(\\"stylesheets\\") : tmpl.stylesheets, _implicitStylesheets);
    }

    __setKey(tmpl, \\"stylesheetTokens\\", {
      hostAttribute: \\"x-cmp1_cmp1-host\\",
      shadowAttribute: \\"x-cmp1_cmp1\\"
    });

    var b = window._ES5ProxyType ? window.get(\\"B\\") : window[\\"B\\"];

    var c = window._ES5ProxyType ? window.get(\\"C\\") : window[\\"C\\"];

    a();
    b();
    c();

    var Cmp1 =
    /*#__PURE__*/
    function (_LightningElement) {
      _inherits(Cmp1, _LightningElement);

      function Cmp1() {
        _classCallCheck(this, Cmp1);

        return _possibleConstructorReturn(this, __callKey2(_getPrototypeOf(Cmp1), \\"apply\\", this, arguments));
      }

      return Cmp1;
    }(lwc.LightningElement);

    var cmp1 = lwc.registerComponent(Cmp1, {
      tmpl: _tmpl
    });

    return cmp1;

});
",
  },
  "descriptor": "component://x/cmp1",
  "uids": Object {
    "compat": "3eaef4ebad",
    "dev": "de273ca667",
    "prod": "f91cadc605",
    "prod_compat": "5d228913fc",
    "prod_debug": "de273ca667",
    "prod_debug_compat": "3eaef4ebad",
  },
}
`;
